lib/spack/spack/cmd/repo.py

```diff
@@ -420,6 +447,66 @@ def repo_set(args):
     tty.msg(f"Updated repo '{namespace}'")
 
 
+def repo_update(args: Any) -> int:
+    config = spack.config.CONFIG
+    scope = args.scope
+
+    descriptors = spack.repo.RepoDescriptors.from_config(
+        spack.repo.package_repository_lock(), config, scope=scope
+    )
+
+    # Get the repos for the specific scope we're modifying
+    scope_repos: Dict[str, Any] = spack.config.get("repos", default={}, scope=args.scope)
+
+    namespace_flags = ["commit", "tag", "branch"]
+    active_flag = next((attr for attr in namespace_flags if getattr(args, attr)), None)
```

Comment:
same for `--remote`?

Reply:
I'd originally had `--remote` in here, but since it defaults to "origin" it's always active which messes up the logic.

Reply:
You can set default=None to distinguish between "not set" and "set to default value"